js/libs/form.js

```diff
@@ -610,6 +610,18 @@ Fliplet.Widget.instance('form-builder', function(data) {
           $form.$forceUpdate();
         }, 10);
 
+        function validateCheckboxValue(value) {
+          if (typeof value === 'string') {
+            value = value.split(',');
```

Comment:
also add a `.map(function (value) { return value.trim(); });`, @tonytlwu do you agree?

Reply:
I agree. Also, is it likely that the value contains `,` as well and might be wrapped in `""`?

Reply:
@tonytlwu 
In case we pass strings separated by commas only first sting will get selected.
![picturemessage_xchd0qgd vpj](https://user-images.githubusercontent.com/52824207/69540326-03665700-0f8f-11ea-85a6-51c783f41fff.png)

In case we pass one string separated inside by commas we select all values.
![picturemessage_ziqcglde qm1](https://user-images.githubusercontent.com/52824207/69540327-03feed80-0f8f-11ea-86bc-c015037db318.png)

Reply:
What if the options for the field were:

- Option 1
- Option 2, with comma
- Option 3

Reply:
Also, why are we calling .val() with multiple parameters?

I can imagine calling `.val(['Option 1', 'Option 2', 'Option 3'])`, not `.val('Option 1', 'Option 2', 'Option 3')`.

Reply:
@tonytlwu If the user will have a comma inside field option then it would not work because we use comma as a separator.  Do you have any ideas on how we should process it?

Reply:
We have a pretty extensively tested function for splitting a string by commas, while ignoring the commas that are wrapped in `"  "`s.

See https://github.com/Fliplet/fliplet-widget-dynamic-lists/blob/aa606369efcba8cd04df0905a8895f5b8269f361/js/utils.js#L130

Reply:
@tonytlwu The function you provided would work in the case if the user wraps option in double-quotes. 
![picturemessage_wrmjmue4 dbn](https://user-images.githubusercontent.com/52824207/69559746-328ebf80-0fb3-11ea-90a2-875d7030c068.png)

I assume that user would set value in the following form .val('Option 1, Option 2,, Option 3') in that case function would still fail.

Reply:
I was commenting on the way that we're using a "split by comma" operation, and that if we're doing this, then we should consider what happens if a comma was meant to be there.

I haven't looked at the wider concept of what this PR is trying to do. For me, a `.val('Foo bar')` or `.set('Foo bar')` would both do the following when it comes to checkboxes:

1. Check if the parameter is an array. If not put it into an array of 1 element.
1. Remove any values in the parameter that isn't an available option for the field
1. Remove duplicates
1. Makes sure that when the field is submitted, the values appear in the order as configured in the available options -- this might be done during submission.

@squallstar Do you have anything to add? You should review and approve the solution is correct.

Reply:
@tonytlwu I agree with your suggested solution 👍

Reply:
fixes

Reply:
@MaksymShokin I'm not sure if this is resolved.

The first line in this function still starts with:

```js
if (!_.isArray(value)) {
  value = value.split(',');
}
```

This has no correlation to the first step I mentioned above:

> 1. Check if the parameter is an array. If not **put it into an array of 1 element**.

If I'm wrong, can you explain how your code relates to what we've outlined ☝️ ?

Reply:
@MaksymShokin I'm not sure if this is resolved.

The first line in this function still starts with:

```js
if (!_.isArray(value)) {
  value = value.split(',');
}
```

This has no correlation to the first step I mentioned above:

> 1. Check if the parameter is an array. If not **put it into an array of 1 element**.

If I'm wrong, can you explain how your code relates to what we've outlined ☝️ ?